doc/install/dextool.def

```diff
@@ -0,0 +1,59 @@
+Bootstrap: docker
+From: ubuntu:18.04
+# This file provides a Singularty image of the packages necessary to run dextool
+# Author: John Tinnerholm 2022-06-14 (Assumes the stable LLVM, which at this time is 13.)
+# For use in the LiU Course TDDD04
+%post -c /bin/bash
+    apt-get -y update
+    apt-get -y install wget
+    apt-get -y install curl
+    apt-get -y install git
+    apt-get -y install software-properties-common
+    apt-get -y update
+    echo "Installing LLVM repositories"
+    echo "Adding LLVM keys..."
+    wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|apt-key add -
+    echo "Keys added!"
+    add-apt-repository ppa:ubuntu-toolchain-r/test
+    add-apt-repository 'deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-13 main'
+    echo "Added llvm repository"
+    apt search llvm-
+    apt search clang-
+    apt search libclang-
+    apt-get -y install llvm-13-dev
+    apt-get -y install build-essential
+    apt-get -y install cmake
+    apt-get -y install llvm-13
+    apt-get -y install llvm-13-dev
+    apt-get -y install clang-13
+    apt-get -y install libclang-13-dev
+    apt-get -y install libsqlite3-dev
+    echo "LLVM Dependencies + SQL Installed."
+    echo "Installing D-Compiler and other auxilary resources"
+    mkdir -p ~/dlang
+    wget https://dlang.org/install.sh -O ~/dlang/install.sh
+    chmod +777 ~/dlang/install.sh
+    ~/dlang/install.sh install dmd-2.097.2
```

Comment:
you only need one compiler. to build the release build i recommend using LDC. its optimizing backend is superior to dmd. it is actually so that dmd for release build may corrupt the final binaries.... happened with some older versions.
thus you could remove all traces of dmd from here.

Reply:
Gotcha!

Reply:
What change do I need to achieve this, I removed dmd. 

got ldc2. 


This results in (Having activated ldc): 

(ldc-1.28.0)Singularity dextool.sif:~/Projects/SingularityPlayground/dextool/build> cmake -CMAKE_D_COMPILER=ldcd2 -DCMAKE_INSTALL_PREFIX=/usr ..
loading initial cache file MAKE_D_COMPILER=ldcd2
CMake Error: Error processing file: MAKE_D_COMPILER=ldcd2
-- Found host D compiler /home/johti17/dlang/ldc-1.28.0/bin/ldmd2, with default flags ''
-- These can be overridden when calling cmake with -D<variable>
-- Host D compiler version: LDC - the LLVM D compiler (1.28.0)
-- Host D compiler linker program: /usr/bin/cc
-- Host D compiler linker flags: -fuse-ld=gold;-L/home/johti17/dlang/ldc-1.28.0/bin/../lib;-lphobos2-ldc;-ldruntime-ldc;-Wl,--gc-sections;-lrt;-ldl;-lpthread;-lm;-m64
-- Using D compiler: /home/johti17/dlang/ldc-1.28.0/bin/ldmd2
-- 0 llvm-config candidates:
Llvm("/usr/bin/llvm-config-13", SemVer([13, 0, 0]))
Using:
/usr/bin/llvm-config-13
-- llvm-config VERSION: LLVM_13_0_0
-- llvm-config MAJOR_VERSION: 13
-- llvm-config LIBDIR: /usr/lib/llvm-13/lib
-- llvm-config LDFLAGS: -L/usr/lib/llvm-13/lib
-- llvm-config INCLUDE: -I/usr/lib/llvm-13/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
-- llvm-config LIBS: -Wl,--enable-new-dtags -Wl,--no-as-needed -Wl,-rpath /usr/lib/llvm-13/lib -lLLVM-13  -lstdc++ -lc -lm
-- clang-config LDFLAGS: -Wl,--enable-new-dtags -Wl,--no-as-needed -L/usr/lib/llvm-13/lib -Wl,-rpath /usr/lib/llvm-13/lib
-- libclang config status : YES
-- libclang libs: -lclangFrontendTool -lclangRewriteFrontend -lclangDynamicASTMatchers -lclangFrontend -lclangASTMatchers -lclangParse -lclangSerialization -lclangRewrite -lclangSema -lclangEdit -lclangAnalysis -lclangAST -lclangLex -lclangBasic -l:libclang.so
-- libclang linker flags: -Wl,--enable-new-dtags -Wl,--no-as-needed -L/usr/lib/llvm-13/lib -Wl,-rpath /usr/lib/llvm-13/lib
-- libLLVM config status: YES
-- libLLVM D flags: -version=LLVM_13_0_0 -version=LLVM_Target_X86
-- libLLVM CXX flags: -I/usr/lib/llvm-13/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
-- libLLVM linker flags: -L/usr/lib/llvm-13/lib
-- libLLVM libs: -Wl,--enable-new-dtags -Wl,--no-as-needed -Wl,-rpath /usr/lib/llvm-13/lib -lLLVM-13  -lstdc++ -lc -lm
-- sqlite3 found: -L/usr/lib/x86_64-linux-gnu -l:libsqlite3.so
-- Configuring incomplete, errors occurred!
See also "/home/johti17/Projects/SingularityPlayground/dextool/build/CMakeFiles/CMakeOutput.log".
(ldc-1.28.0)Singularity dextool.sif:~/Projects/SingularityPlayground/dextool/build>

Do you have any clue on this should I open a separate issue?

Reply:
Sorry misspelling by me above: 

Full output with the cache of cmake reset: 

```
(ldc-1.28.0)Singularity dextool.sif:~/Projects/SingularityPlayground/dextool/build> cmake -CMAKE_D_COMPILER=ldc2 -DCMAKE_INSTALL_PREFIX=/usr ..
loading initial cache file MAKE_D_COMPILER=ldc2
CMake Error: Error processing file: MAKE_D_COMPILER=ldc2
-- 0 llvm-config candidates:
Llvm("/usr/bin/llvm-config-13", SemVer([13, 0, 0]))
Using:
/usr/bin/llvm-config-13
-- llvm-config VERSION: LLVM_13_0_0
-- llvm-config MAJOR_VERSION: 13
-- llvm-config LIBDIR: /usr/lib/llvm-13/lib
-- llvm-config LDFLAGS: -L/usr/lib/llvm-13/lib
-- llvm-config INCLUDE: -I/usr/lib/llvm-13/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
-- llvm-config LIBS: -Wl,--enable-new-dtags -Wl,--no-as-needed -Wl,-rpath /usr/lib/llvm-13/lib -lLLVM-13  -lstdc++ -lc -lm
-- clang-config LDFLAGS: -Wl,--enable-new-dtags -Wl,--no-as-needed -L/usr/lib/llvm-13/lib -Wl,-rpath /usr/lib/llvm-13/lib
Detected user configuration of CLANG
Detected user configuration of LLVM
-- libclang config status : YES
-- libclang libs: -lclangFrontendTool -lclangRewriteFrontend -lclangDynamicASTMatchers -lclangFrontend -lclangASTMatchers -lclangParse -lclangSerialization -lclangRewrite -lclangSema -lclangEdit -lclangAnalysis -lclangAST -lclangLex -lclangBasic -l:libclang.so
-- libclang linker flags: -Wl,--enable-new-dtags -Wl,--no-as-needed -L/usr/lib/llvm-13/lib -Wl,-rpath /usr/lib/llvm-13/lib
-- libLLVM config status: YES
-- libLLVM D flags: -version=LLVM_13_0_0 -version=LLVM_Target_X86
-- libLLVM CXX flags: -I/usr/lib/llvm-13/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
-- libLLVM linker flags: -L/usr/lib/llvm-13/lib
-- libLLVM libs: -Wl,--enable-new-dtags -Wl,--no-as-needed -Wl,-rpath /usr/lib/llvm-13/lib -lLLVM-13  -lstdc++ -lc -lm
-- sqlite3 found: -L/usr/lib/x86_64-linux-gnu -l:libsqlite3.so
-- Configuring incomplete, errors occurred!
See also "/home/johti17/Projects/SingularityPlayground/dextool/build/CMakeFiles/CMakeOutput.log".
(ldc-1.28.0)Singularity dextool.sif:~/Projects/SingularityPlayground/dextool/build> ls
CMakeCache.txt  CPackConfig.cmake        cmakeExtractDMDSystemLinker    cmakeExtractDMDSystemLinker.o  cmake_introspect_llvm.o  plugin     source   symlink.o
CMakeFiles      CPackSourceConfig.cmake  cmakeExtractDMDSystemLinker.d  cmake_introspect_llvm          libs                     resources  symlink  vendor
(ldc-1.28.0)Singularity dextool.sif:~/Projects/SingularityPlayground/dextool/build> rm CMakeCache.txt
(ldc-1.28.0)Singularity dextool.sif:~/Projects/SingularityPlayground/dextool/build> cmake -CMAKE_D_COMPILER=ldc2 -DCMAKE_INSTALL_PREFIX=/usr ..
loading initial cache file MAKE_D_COMPILER=ldc2
CMake Error: Error processing file: MAKE_D_COMPILER=ldc2
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found host D compiler /home/johti17/dlang/ldc-1.28.0/bin/ldmd2, with default flags ''
-- These can be overridden when calling cmake with -D<variable>
-- Host D compiler version: LDC - the LLVM D compiler (1.28.0)
-- Host D compiler linker program: /usr/bin/cc
-- Host D compiler linker flags: -fuse-ld=gold;-L/home/johti17/dlang/ldc-1.28.0/bin/../lib;-lphobos2-ldc;-ldruntime-ldc;-Wl,--gc-sections;-lrt;-ldl;-lpthread;-lm;-m64
-- Using D compiler: /home/johti17/dlang/ldc-1.28.0/bin/ldmd2
-- 0 llvm-config candidates:
Llvm("/usr/bin/llvm-config-13", SemVer([13, 0, 0]))
Using:
/usr/bin/llvm-config-13
-- llvm-config VERSION: LLVM_13_0_0
-- llvm-config MAJOR_VERSION: 13
-- llvm-config LIBDIR: /usr/lib/llvm-13/lib
-- llvm-config LDFLAGS: -L/usr/lib/llvm-13/lib
-- llvm-config INCLUDE: -I/usr/lib/llvm-13/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
-- llvm-config LIBS: -Wl,--enable-new-dtags -Wl,--no-as-needed -Wl,-rpath /usr/lib/llvm-13/lib -lLLVM-13  -lstdc++ -lc -lm
-- clang-config LDFLAGS: -Wl,--enable-new-dtags -Wl,--no-as-needed -L/usr/lib/llvm-13/lib -Wl,-rpath /usr/lib/llvm-13/lib
-- libclang config status : YES
-- libclang libs: -lclangFrontendTool -lclangRewriteFrontend -lclangDynamicASTMatchers -lclangFrontend -lclangASTMatchers -lclangParse -lclangSerialization -lclangRewrite -lclangSema -lclangEdit -lclangAnalysis -lclangAST -lclangLex -lclangBasic -l:libclang.so
-- libclang linker flags: -Wl,--enable-new-dtags -Wl,--no-as-needed -L/usr/lib/llvm-13/lib -Wl,-rpath /usr/lib/llvm-13/lib
-- libLLVM config status: YES
-- libLLVM D flags: -version=LLVM_13_0_0 -version=LLVM_Target_X86
-- libLLVM CXX flags: -I/usr/lib/llvm-13/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
-- libLLVM linker flags: 
[CMakeOutput.log](https://github.com/joakim-brannstrom/dextool/files/8909053/CMakeOutput.log)
-L/usr/lib/llvm-13/lib
-- libLLVM libs: -Wl,--enable-new-dtags -Wl,--no-as-needed -Wl,-rpath /usr/lib/llvm-13/lib -lLLVM-13  -lstdc++ -lc -lm
-- sqlite3 found: -L/usr/lib/x86_64-linux-gnu -l:libsqlite3.so
-- Configuring incomplete, errors occurred!
See also "/home/johti17/Projects/SingularityPlayground/dextool/build/CMakeFiles/CMakeOutput.log"
```

Reply:
I think the cmake command is wrong. Missing a "D". Should be:
```bash
cmake -DCMAKE_D_COMPILER=ldc2 -DCMAKE_INSTALL_PREFIX=/usr ..
```

but it shouldn't be necessary to specify because ldc is preferred over dmd and if only ldc is found in `$PATH` then it is chosen as the compiler.